In [5]:
import numpy as np
import pandas as pd
import scipy as sp
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.formula.api import ols

In [6]:
# Import plotting routines
%matplotlib inline
import matplotlib.pyplot as plt
import matplotlib
matplotlib.style.use('ggplot')

In [7]:
emdat = pd.read_csv('EM-DAT/Storms_all_countries.csv')
emdat

year disaster type  iso  \
0     1900         Storm  USA   
1     1902         Storm  MMR   
2     1903         Storm  JAM   
3     1903         Storm  USA   
4     1904         Storm  BGD   
5     1905         Storm  PHL   
6     1906         Storm  HKG   
7     1906         Storm  USA   
8     1909         Storm  BGD   
9     1909         Storm  HTI   
10    1909         Storm  USA   
11    1910         Storm  USA   
12    1911         Storm  BGD   
13    1911         Storm  TWN   
14    1912         Storm  CAN   
15    1912         Storm  CHN   
16    1912         Storm  JAM   
17    1912         Storm  JPN   
18    1912         Storm  PHL   
19    1913         Storm  USA   
20    1914         Storm  TKL   
21    1915         Storm  HTI   
22    1915         Storm  USA   
23    1916         Storm  IND   
24    1917         Storm  JAM   
25    1917         Storm  JPN   
26    1918         Storm  USA   
27    1919         Storm  BGD   
28    1919         Storm  USA   
29    1920         Storm  USA   
...    ...           ...  ...   
1947  2015         Storm  DOM   
1948  2015         Storm  SLV   
1949  2015         Storm  HTI   
1950  2015         Storm  IND   
1951  2015         Storm  IRL   
1952  2015         Storm  ISR   
1953  2015         Storm  JPN   
1954  2015         Storm  KIR   
1955  2015         Storm  LBN   
1956  2015         Storm  MDG   
1957  2015         Storm  MWI   
1958  2015         Storm  MEX   
1959  2015         Storm  FSM   
1960  2015         Storm  MMR   
1961  2015         Storm  MNP   
1962  2015         Storm  PSE   
1963  2015         Storm  PRY   
1964  2015         Storm  PHL   
1965  2015         Storm  SLB   
1966  2015         Storm  SOM   
1967  2015         Storm  SYR   
1968  2015         Storm  TWN   
1969  2015         Storm  TZA   
1970  2015         Storm  TUV   
1971  2015         Storm  GBR   
1972  2015         Storm  USA   
1973  2015         Storm  VUT   
1974  2015         Storm  VNM   
1975  2015         Storm  YEM   
1976  2015         Storm  ZWE   

                                           country_name  occurrence  \
0                        United States of America (the)           1   
1                                               Myanmar           1   
2                                               Jamaica           1   
3                        United States of America (the)           1   
4                                            Bangladesh           1   
5                                     Philippines (the)           1   
6                                             Hong Kong           1   
7                        United States of America (the)           2   
8                                            Bangladesh           2   
9                                                 Haiti           1   
10                       United States of America (the)           2   
11                       United States of America (the)           1   
12                                           Bangladesh           1   
13                           Taiwan (Province of China)           1   
14                                               Canada           1   
15                                                China           1   
16                                              Jamaica           1   
17                                                Japan           1   
18                                    Philippines (the)           1   
19                       United States of America (the)           1   
20                                              Tokelau           1   
21                                                Haiti           1   
22                       United States of America (the)           1   
23                                                India           1   
24                                              Jamaica           1   
25                                                Japan           1   
26                       United States of 

In [8]:
emdat[['Total affected','Total damage','Total deaths']].dropna(axis=0)

Total affected  Total damage  Total deaths
14            2700.0        5000.0          28.0
63           12000.0       40000.0        2000.0
114        1881672.0      400000.0        4197.0
127            300.0       18000.0          14.0
145          20200.0       56000.0         154.0
152            800.0        2000.0          23.0
155            452.0       50000.0        1014.0
207        1500000.0      600000.0        5264.0
216           1250.0       35000.0           5.0
225         425000.0       32000.0         631.0
238           8000.0         500.0           2.0
251        1000000.0       50000.0       22079.0
266            800.0         250.0         300.0
271          80000.0       10000.0         100.0
275          86830.0        1000.0         210.0
276         400000.0        4100.0         450.0
277            518.0         300.0           1.0
278        1225546.0       43000.0         636.0
280         380000.0       37300.0         206.0
283         700000.0       50000.0        7000.0
285       15660000.0       57700.0       48921.0
291         500000.0        1000.0         100.0
295           1000.0     1610000.0         570.0
299              3.0       15500.0           5.0
300        1800000.0       22400.0         850.0
304         174030.0       18000.0           5.0
305          12942.0        5000.0          74.0
306          10000.0       70000.0          25.0
307          67881.0       20000.0         480.0
309         150000.0        4200.0          18.0
...              ...           ...           ...
1899       1963444.0     1474230.0          79.0
1905            40.0      545000.0           1.0
1907      11209694.0    10069973.0         329.0
1911             1.0      400000.0           8.0
1913        931564.0     7000000.0         111.0
1916         41282.0     6541000.0         129.0
1917           101.0       11000.0          24.0
1921        144135.0     2539500.0          19.0
1922        117000.0      300000.0          49.0
1926      13068983.0     1062899.0         326.0
1928           265.0       85000.0           1.0
1933          4014.0       31000.0           1.0
1934         18000.0      100000.0           5.0
1935          1232.0    10366000.0         139.0
1937         48075.0       10700.0          38.0
1938          4500.0       17200.0           8.0
1939          5938.0     1653000.0           4.0
1940          6710.0      100000.0          35.0
1945       2058700.0     7435690.0          58.0
1946         28594.0      482810.0          30.0
1950        135100.0      161000.0          92.0
1953         61033.0      223200.0          11.0
1956        182437.0       46000.0          95.0
1958         14128.0      410000.0          43.0
1964       3606205.0     1881367.0         148.0
1968           762.0       26000.0           9.0
1971         15600.0     1100000.0           3.0
1972           372.0    10825000.0         175.0
1973        166000.0      449400.0          11.0
1975        125065.0      100000.0          32.0

[640 rows x 3 columns]

Notes: EMDAT summarizes events per year. This means that one datapoint can contain the outcome of more than one storm. It is hence not possible to assign storm effects or deaths to only singular events, unless there was only one storm in place.
<br>

The current selection contains 1976 entries, but only 174 of these are complete in all columns. <br>

When looking at total affected and total damage, the dataset contains 719 complete entries <br>

When looking at total affected and total damage and total death, the dataset contains 640 complete entries <br>

In [9]:
emdat.rename(columns={' country_name':'country','Total affected':'tot_aff','Total damage':'tot_dam','Total deaths':'tot_death'},inplace=True)

In [10]:
emdat.head(2)

year disaster type  iso                         country  occurrence  \
0  1900         Storm  USA  United States of America (the)           1   
1  1902         Storm  MMR                         Myanmar           1   

   tot_death  Affected  Injured  Homeless  tot_aff  tot_dam  
0     6000.0       NaN      NaN       NaN      NaN  30000.0  
1      600.0       NaN      NaN       NaN      NaN      NaN

In [11]:
storms = emdat[emdat['disaster type']=='Storm']

In [12]:
storms.head()

year disaster type  iso                         country  occurrence  \
0  1900         Storm  USA  United States of America (the)           1   
1  1902         Storm  MMR                         Myanmar           1   
2  1903         Storm  JAM                         Jamaica           1   
3  1903         Storm  USA  United States of America (the)           1   
4  1904         Storm  BGD                      Bangladesh           1   

   tot_death  Affected  Injured  Homeless  tot_aff  tot_dam  
0     6000.0       NaN      NaN       NaN      NaN  30000.0  
1      600.0       NaN      NaN       NaN      NaN      NaN  
2       65.0       NaN      NaN       NaN      NaN      NaN  
3       98.0       NaN      NaN       NaN      NaN      NaN  
4        NaN       NaN      NaN       NaN      NaN      NaN

In [14]:
storms = storms[['year','iso','country','occurrence','tot_aff','tot_death','tot_dam']]

In [15]:
storms.head()

year  iso                         country  occurrence  tot_aff  tot_death  \
0  1900  USA  United States of America (the)           1      NaN     6000.0   
1  1902  MMR                         Myanmar           1      NaN      600.0   
2  1903  JAM                         Jamaica           1      NaN       65.0   
3  1903  USA  United States of America (the)           1      NaN       98.0   
4  1904  BGD                      Bangladesh           1      NaN        NaN   

   tot_dam  
0  30000.0  
1      NaN  
2      NaN  
3      NaN  
4      NaN

In [16]:
regress_multiple_p = smf.ols(formula='tot_aff ~ tot_dam + tot_death-1', data=storms).fit()
regress_multiple_p.params
regress_multiple_p.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                tot_aff   R-squared:                       0.016
Model:                            OLS   Adj. R-squared:                  0.013
Method:                 Least Squares   F-statistic:                     5.091
Date:                Mon, 02 May 2016   Prob (F-statistic):            0.00640
Time:                        15:51:35   Log-Likelihood:                -10911.
No. Observations:                 640   AIC:                         2.183e+04
Df Residuals:                     638   BIC:                         2.184e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
tot_dam        0.0603      0.030      2.023      0.044         0.002     0.119
tot_death     40.9474     16.873      2.427      0.016         7.814    74.081
==============================================================================
Omnibus:                     1118.989   Durbin-Watson:                   1.967
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           750435.165
Skew:                          11.070   Prob(JB):                         0.00
Kurtosis:                     169.286   Cond. No.                         566.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [17]:
#x_grid=np.linspace(0,storms['tot_aff'].max(), 100)
#y_pol_intercept = regress_multiple_p.params[0] + regress_multiple_p.params[1]*x_grid + regress_multiple_p.params[2]*x_grid
#y_pol= regress_multiple_p.params[0]*x_grid + regress_multiple_p.params[1]*x_grid

In [18]:
%matplotlib inline

In [19]:
%matplotlib qt

In [20]:
fig = plt.figure(figsize=[12,9])
plt.scatter(storms['tot_aff'], storms['tot_dam'], color='k', clip_on=False)
#plt.plot(storms['tot_aff'],  regress_multiple_p.params[0]*storms['tot_dam']
#        + regress_multiple_p.params[1] * storms['tot_death'], 'rx')
plt.plot(storms['tot_aff'],  regress_multiple_p.params[0]*storms['tot_aff']
         + regress_multiple_p.params[1]*storms['tot_death'], 'rx')


#plt.plot(x_grid, y_pol)
# for k in df_stats.index.values:
#     plt.annotate(xy = [df_stats.loc[k,'social_p'], df_stats.loc[k,'CPI2015']],
#                  s=k, xycoords='data')

plt.xlabel('Total affected')
plt.ylabel('Total damage')
plt.title('Storms')
plt.xlim([0,2e7])
plt.ylim([0,1e7])
plt.legend()

In [21]:
fig = plt.figure(figsize=[12,9])
plt.scatter(storms['tot_aff'], storms['tot_death'], color='k', clip_on=False)

plt.plot(storms['tot_aff'],  regress_multiple_p.params[0]*storms['tot_aff']
         + regress_multiple_p.params[1]*storms['tot_death'], 'rx')

#plt.plot(x_grid, y_pol)
# for k in df_stats.index.values:
#     plt.annotate(xy = [df_stats.loc[k,'social_p'], df_stats.loc[k,'CPI2015']],
#                  s=k, xycoords='data')

plt.xlabel('Total affected')
plt.ylabel('Total deaths')
plt.title('Storms')
plt.xlim([0,2e7])
plt.ylim([0,1e4])
plt.legend()